### PR TITLE
feat: add wasm compatibility to tailcall/core

### PR DESCRIPTION
**Pull request by ssddOnTop**
Orignal issue: #750

### CODE REVIEW

**Comment by ssddOnTop** [comment]
List of things which are **not supported** in this PR:

## GRPC
**Cause**

Currently we only support compiling proto files with fs. [code referance](https://github.com/tailcallhq/tailcall/blob/2dcacb6e8fc48b840dbd4e2087e052f316df23ca/src/grpc/protobuf.rs#L80)

**Possible Solution**

Add support for loading proto files over http. 

## Caching

**Cause** 

reqwest::Response is not thread safe and it lacks support for `try_from` trait for standard http::Response. However this feature is expected to be released [soon](https://github.com/seanmonstar/reqwest/pull/1920#issuecomment-1868505500)

File: src/http/client/client_cli.rs [hunk 1]
```diff
@@ -4,27 +4,18 @@ use http_cache_reqwest::{Cache, CacheMode, HttpCache, HttpCacheOptions, MokaMana
...
 use reqwest_middleware::{ClientBuilder, ClientWithMiddleware};
 
-use super::Response;
-use crate::config::{self, Upstream};
-
-#[async_trait::async_trait]
-pub trait HttpClient: Sync + Send {
-  async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response>;
-  async fn execute_raw(&self, req: reqwest::Request) -> anyhow::Result<reqwest::Response>;
-}
```
Comment (tusharmath) [inline src/http/client/client_cli.rs:14]: Make Response polymorphic on Body

```rust
pub trait HttpClient: Sync + Send {
  async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response<Value>>;
  async fn execute_raw(&self, req: reqwest::Request) -> anyhow::Result<reqwest::Response<Vec<u8>>>;
}
```

**Comment by tusharmath** [comment]
**IMPORTANT**
Add a wasm build only for cloudflare before merging this PR.

File: Cargo.toml [hunk 2]
```diff
@@ -8,63 +8,60 @@ name = "tailcall"
...
 num_cpus = "1.16.0"
-hyper-rustls = "0.24.2"
-rustls = "0.21.9"
-rustls-pemfile = "1.0.4"
 fnv = "1.0.7"
-futures-channel = { version = "0.3.30" }
+futures-channel = { version = "0.3.29" }
 futures-timer = { version = "3.0.2" }
-futures-util = { version = "0.3.30" }
+futures-util = { version = "0.3.29" }
```
Comment (tusharmath) [inline Cargo.toml:60]: don't downgrade dependencies.

**Pull request merged**